Source/Plugins/NetworkEvents/NetworkEvents.cpp

```
@@ -238,10 +245,69 @@ String NetworkEvents::handleSpecialMessages(const String& s)
         status += CoreServices::RecordNode::getExperimentNumber();
         return status;
     }
+    else if (cmd.compareIgnoreCase("TTL") == 0)
+    {
+        std::cout << "into ttl event case" << std::endl;
+        // Default to chan 0 and off
+        int currEventChannel = 0;
+        bool onOff = false;
+        /** First set optional parameters (name/value pairs)*/
+        if (s.contains("="))
+        {
+            String params = s.substring(cmd.length()).trim();
+            StringPairArray dict = parseNetworkMessage(params);
+
+            StringArray keys = dict.getAllKeys();
+            for (int i = 0; i < keys.size(); ++i)
+            {
+                String key = keys[i];
+                String value = dict[key];
+
+                if (key.compareIgnoreCase("EventChannel") == 0)
+                {
+                    std::cout << value << std::endl;
+                    currEventChannel = value.getIntValue();
+                }
+                else if (key.compareIgnoreCase("onOff") == 0)
+                {
+                    std::cout << value << std::endl;
+                    if (value.compareIgnoreCase("on") == 0)
```

This definitely works, but a couple preferences:

* Since whether the event is turning on or off is really a boolean, I'd prefer the value to be just `"1"` or `"0"` (like for `"CreateNewDir"` above) rather than "on" or "off". We could call the key `On` or `State` (which matches the `getState` method of `TTLEvent`).

* `"EventChannel"` could probably just be `"Channel"`.

* We should return an error message and not trigger an event if the channel is < 0 or >= 8. Or maybe we want to use 1-based channel indices instead here; it might be less confusing as a user since the GUI always numbers channels from 1 to 8. Come to think of it, we should change the Event Broadcaster to also output 1-based channel numbers.